satpy/readers/seviri_l2_bufr.py

```diff
@@ -54,8 +54,19 @@ def __init__(self, filename, filename_info, filetype_info, **kwargs):
                                                       filename_info,
                                                       filetype_info)
 
-        self.filename = filename
-        self.mpef_header = self._read_mpef_header()
+        if ('server' in filename_info):
+            # EUMETSAT Offline Bufr product
+            self.mpef_header = self._read_mpef_header()
+        else:
+            # Product was retrieved from the EUMETSAT Data Centre
+            timeStr = self.get_attribute('typicalDate')+self.get_attribute('typicalTime')
+            buf_start_time = datetime.strptime(timeStr, "%Y%m%d%H%M%S")
+            sc_id = self.get_attribute('satelliteIdentifier')
+            self.mpef_header = {}
+            self.mpef_header['NominalTime'] = buf_start_time
+            self.mpef_header['SpacecraftName'] = data_centre_dict[sc_id]['name']
+            self.mpef_header['RectificationLongitude'] = data_centre_dict[sc_id]['ssp']
```

Comment:
Should this be passed to the dataset array as metadata ?

Reply:
Or is it already ?

Reply:
spacecraftname and Rectification and set as metadata already 

xarr.attrs['spacecraft_name'] = self.spacecraft_name
xarr.attrs['ssp_lon'] = self.ssp_lon

start time is a scene attribute so i guess that is also ok - Sauli?

Reply:
looks good to me.

Reply:
Perfect then